internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -443,24 +450,32 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	err := c.daemonReload(ctx)
+	if err != nil &&
+		(errors.Is(err, context.DeadlineExceeded) ||
+			errors.Is(err, context.Canceled)) {
+		return fmt.Errorf("could not reload deamon: %w", err)
+	}
 	if err == nil {
 		return nil
 	}
 
 	signal := make(chan struct{})
 	backExp := backoff.NewExpBackoff(signal, 10*time.Second, 1*time.Minute)
 
-	for i := 5; i >= 0; i-- {
+	var i int
+	for ; i < 5; i++ {
 		backExp.Wait()
 		c.log.Info("Retrying to restart...")
 		err = c.daemonReload(ctx)
-		if err == nil {
+		if err == nil ||
+			errors.Is(err, context.DeadlineExceeded) ||
+			errors.Is(err, context.Canceled) {
```

Comment:
You could replace this with https://pkg.go.dev/github.com/cenkalti/backoff/v4 which has a `WithContext` option to natively understands contexts. We already use this package for upgrades. Example:

https://github.com/cmacknz/elastic-agent/blob/261b94e5ec8372dda376ac80cdd212902b3a5aba/internal/pkg/agent/application/upgrade/step_download.go#L155-L185